airqo-platform/AirQo-frontend
# [NetManager] Merge url configs and use API-gateway urls

#### Summary of Changes (What does this PR do?)
- Removes the `constants.js` file.
- Each micro-service has a dedicated file to hold it's urls
- Instead of the previous urls settings (having hardcoded urls for each environment e.g staging, prod), the new approach separates urls by micro-service. The urls share a fallback `REACT_APP_BASE_URL` base url incase the base url for a given micro-service is not set. This approach allows for much flexibility during the development process without necessarily duplicating the endpoints (urls).
- The new `BASE_URL` env keys;
 -- `REACT_APP_BASE_URL` (required)
 -- `REACT_APP_BASE_ANALYTICS_URL` (optional)
 -- `REACT_APP_BASE_AUTH_SERVICE_URL` (optional)
 -- `REACT_APP_BASE_DATA_MANAGEMENT_URL` (optional)
 -- `REACT_APP_BASE_DEVICE_MONITORING_URL` (optional)
 -- `REACT_APP_BASE_DEVICE_REGISTRY_URL` (optional)
 -- `REACT_APP_BASE_LOCATE_URL` (optional)
 -- `REACT_APP_BASE_LOCATION_REGISTRY_URL` (optional)
 -- `REACT_APP_BASE_PREDICT_URL` (optional)

#### Is this change ready to hit production in its current state?

#### Status of maturity (all need to be checked before merging):


#### How should this be manually tested?
* Pull and checkout to this [branch](https://github.com/airqo-platform/AirQo-frontend/tree/ch-merge-configs-PLAT-117) locally
* Install node requirements `npm install`
* Start the application `npm run prod-mac` (on mac) or `npm run prod-pc` (on windows platform)
*  Setup the `BASE_URL` in `.env` file (Mostly the required key `REACT_APP_BASE_URL`)
* You should be able to hit all the api services (in other words, the app should load all the data without any errors)

#### What are the relevant tickets?
- [PLAT-117](https://airqoteam.atlassian.net/browse/PLAT-117)
- [PLAT-418](https://airqoteam.atlassian.net/browse/PLAT-418)

#### Screenshots (optional)

> Please add run command for Windows too.
